clients/client/src/main/java/org/projectnessie/client/api/CreateReferenceBuilder.java

```diff
@@ -29,10 +29,23 @@
  * @since {@link NessieApiV1}
  */
 public interface CreateReferenceBuilder {
+  /**
+   * Configure source reference name.
```

Comment:
Is this more information than what the method name itself already says?

Reply:
It would be good to at the very least include the docs from the REST API and maybe have some code examples of how to use this CreateReferenceBuilder API for different use cases

Reply:
Suggestion: 
```
Sets the name of the reference what contains the hash of the reference-to-be-created.

This reference name will be used for authorization purposes to validate that the caller has read access to the provided hash.

If not explicitly set, the default branch will be used as the source reference name.

The name and hash of the reference to be created is set via {@link #reference()}.
```